### PR TITLE
Update to brod 3.0

### PR DESCRIPTION
**Pull request by id**

### CODE REVIEW

File: web/controllers/api/v1/message_controller.ex [hunk 1]
```diff
@@ -104,46 +105,50 @@ defmodule Kastlex.API.V1.MessageController do
...
+        {:ok, resp}
     end
   end
 
+  defp transform_kafka_message(raw_msg) do
+    msg = kafka_message(raw_msg)
+    %{key: undef_to_nil(msg[:key]),
+      value: undef_to_nil(msg[:value]),
+      offset: msg[:offset],
+      crc: msg[:crc]
```
Comment (zmstone) [inline web/controllers/api/v1/message_controller.ex]: `ts` and `ts_type` missing ?

Comment (id) [inline web/controllers/api/v1/message_controller.ex]: yes, don't want do break users who are already using v1 message api, see v2

File: web/controllers/api/v1/message_controller.ex [hunk 2]
```diff
@@ -80,16 +80,17 @@ defmodule Kastlex.API.V1.MessageController do
...
-                fn(max_bytes_try) ->
-                  :kpro.fetch_request(topic, partition, offset, max_wait_time, min_bytes, max_bytes_try)
-                end
-                resp = do_fetch(pid, req_fun, max_bytes, offset, type)
-                respond(conn, resp, type)
-              {:error, :no_offsets} ->
-                send_json(conn, 404, %{error: "No messages at the requested offset #{orig_offset}"})
-              other ->
-                Logger.error "Cannot resolve logical offset #{orig_offset}: #{inspect other}"
+                request = :kpro.fetch_request(_vsn = 0,
```
Comment (zmstone) [inline web/controllers/api/v1/message_controller.ex]: call `brod_utils:make_fetch_fun/6` instead.

Comment (id) [inline web/controllers/api/v1/message_controller.ex]: fixed for v2

Comment (id) [inline web/controllers/api/v1/message_controller.ex]: fetch fun does not return high wm offset

Comment (id) [inline web/controllers/api/v1/message_controller.ex]: not sure if we really need it, but it's already used in v1

File: mix.lock [hunk 3]
```diff
@@ -1,31 +1,31 @@
 %{"base64url": {:hex, :base64url, "0.0.1", "36a90125f5948e3afd7be97662a1504b934dd5dac78451ca6e9abf85a10286be", [], [], "hexpm"},
   "briefly": {:hex, :briefly, "0.3.0", "16e6b76d2070ebc9cbd025fa85cf5dbaf52368c4bd896fb482b5a6b95a540c2f", [:mix], [], "hexpm"},
-  "brod": {:hex, :brod, "2.5.0", "dd9dd6eb4849a0dbb2aca7ba4d8d423228d17e8822312b10f80b716a150f18ab", [:make, :rebar, :rebar3], [{:kafka_protocol, "0.9.2", [hex: :kafka_protocol, repo: "hexpm", optional: false]}, {:supervisor3, "1.1.5", [hex: :supervisor3, repo: "hexpm", optional: false]}], "hexpm"},
+  "brod": {:hex, :brod, "3.0.0", "b3acc83fcb9a9d23e3dd662766953a42c0f903723b943474eaadf45e6b31026a", [:make, :rebar, :rebar3], [{:kafka_protocol, "1.0.0", [hex: :kafka_protocol, repo: "hexpm", optional: false]}, {:supervisor3, "1.1.5", [hex: :supervisor3, repo: "hexpm", optional: false]}], "hexpm"},
```
Comment (zmstone) [inline mix.lock]: 3.3.0 available now

**Pull request merged**